# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/crypto_hash/crypto_hash.h
# Commit: 2b647eb982bc
# Full Hash: 2b647eb982bc0068d39fe58c6f59bebe55909343
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2023-06-07 21:43:58
# Regressor Bug: 1834862
# File Overlap Count: 1
# Description:
#   Bug 1834862 - implement nsICryptoHash in rust r=jschanck,supply-chain-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D179727
# ==============================================================================

diff -r 7c11166969ff -r 2b647eb982bc security/manager/ssl/crypto_hash/crypto_hash.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/security/manager/ssl/crypto_hash/crypto_hash.h	Wed Jun 07 15:56:37 2023 +0000
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "nsICryptoHash.h"
+
+extern "C" {
+nsresult crypto_hash_constructor(REFNSIID iid, void** result);
+};
+
+nsresult NS_NewCryptoHash(uint32_t aHashType, nsICryptoHash** aOutHasher) {
+  MOZ_ASSERT(aOutHasher);
+
+  nsCOMPtr<nsICryptoHash> hasher;
+  nsresult rv =
+      crypto_hash_constructor(NS_ICRYPTOHASH_IID, getter_AddRefs(hasher));
+  if (NS_FAILED(rv)) {
+    return rv;
+  }
+  rv = hasher->Init(aHashType);
+  if (NS_FAILED(rv)) {
+    return rv;
+  }
+  hasher.forget(aOutHasher);
+
+  return NS_OK;
+}
+
+nsresult NS_NewCryptoHash(const nsACString& aHashType,
+                          nsICryptoHash** aOutHasher) {
+  MOZ_ASSERT(aOutHasher);
+
+  nsCOMPtr<nsICryptoHash> hasher;
+  nsresult rv =
+      crypto_hash_constructor(NS_ICRYPTOHASH_IID, getter_AddRefs(hasher));
+  if (NS_FAILED(rv)) {
+    return rv;
+  }
+  rv = hasher->InitWithString(aHashType);
+  if (NS_FAILED(rv)) {
+    return rv;
+  }
+  hasher.forget(aOutHasher);
+
+  return NS_OK;
+}